x/cash/handler.go

```diff
@@ -37,66 +37,38 @@ func NewSendHandler(auth x.Authenticator, control Controller) SendHandler {
 
 // Check just verifies it is properly formed and returns
 // the cost of executing it
-func (h SendHandler) Check(ctx weave.Context, store weave.KVStore,
-	tx weave.Tx) (weave.CheckResult, error) {
-
-	// ensure type and validate...
-	var res weave.CheckResult
-	rmsg, err := tx.GetMsg()
-	if err != nil {
-		return res, err
-	}
-	msg, ok := rmsg.(*SendMsg)
-	if !ok {
-		return res, errors.WithType(errors.ErrInvalidMsg, rmsg)
-	}
-
-	err = msg.Validate()
-	if err != nil {
-		return res, err
+func (h SendHandler) Check(ctx weave.Context, store weave.KVStore, tx weave.Tx) (weave.CheckResult, error) {
+	var msg SendMsg
```

Comment:
This is cleaner code, isn't it?

:smile:

Reply:
I think it is really nice. It is good that it is so similar to `json.Unmarshal` - any new contributor should find this a familiar construct.